### PR TITLE
<SP-11>: Created Repository level and pom updated

### PR DESCRIPTION
**Pull request by Viacheslav-Pavliuk**
- Created entities
- Created repositories
- Set uped flyway
- Generated migration
- Created tests for repositories

### CODE REVIEW

File: src/main/java/com/summer_prictice/demo/ent/Warehouse.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    public Set<ShipTo> getShipTos() {
+        return shipTos;
+    }
+
+    public void setShipTos(Set<ShipTo> shipTos) {
+        this.shipTos = shipTos;
+    }
+
+
```
Comment (metlaivan) [inline src/main/java/com/summer_prictice/demo/ent/Warehouse.java:47]: No need an empty lines. Please use the formatting for all files.

File: src/main/java/com/summer_prictice/demo/repos/ShipToRepo.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.summer_prictice.demo.repos;
+
+import com.summer_prictice.demo.ent.ShipTo;
+import org.springframework.data.repository.CrudRepository;
+
+public interface ShipToRepo extends CrudRepository<ShipTo,Long> {
```
Comment (metlaivan) [inline src/main/java/com/summer_prictice/demo/repos/ShipToRepo.java:6]: Please use the same name for class names in repository level.

File: src/test/java/com/summer_prictice/demo/SummerPricticeApplicationTests.java [hunk 3]
```diff
@@ -6,6 +6,8 @@
 @SpringBootTest
 class SummerPricticeApplicationTests {
 
+
```
Comment (metlaivan) [inline src/test/java/com/summer_prictice/demo/SummerPricticeApplicationTests.java:9]: no need empty lines.

File: src/test/java/com/summer_prictice/demo/Testing_ShipTo.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.time.LocalDate;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class Testing_ShipTo {
```
Comment (metlaivan) [inline src/test/java/com/summer_prictice/demo/Testing_ShipTo.java:20]: Use class name format

File: src/test/java/com/summer_prictice/demo/Testing_ShipTo.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        st.setUpdatedby("Less");
+        st.setName("Cars");
+        st.setAddress("Suite 12");
+        st.setWarehouseId(wh);
+        sTR.save(st);
+        assertEquals(sTR.findById(Long.valueOf("4")).get().getCreatedat(),date_cr);
+        assertTrue(sTR.findById(Long.valueOf("4")).isPresent());
+
+    }
+    @Test
```
Comment (metlaivan) [inline src/test/java/com/summer_prictice/demo/Testing_ShipTo.java:43]: please format the class

File: src/test/java/com/summer_prictice/demo/Testing_ShipTo.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+public class Testing_ShipTo {
+    @Autowired
+    private WarehouseRepo WhR;
+    @Autowired
+    private ShipToRepo sTR;
+    @Test
+    public void addShipTo(){
+        ShipTo st=new ShipTo();
+        Warehouse wh = WhR.findById(Long.valueOf("3")).get();
+        LocalDate date_cr = LocalDate.of(2011,11,11);
```
Comment (metlaivan) [inline src/test/java/com/summer_prictice/demo/Testing_ShipTo.java:29]: field name must be in Java style

File: src/test/java/com/summer_prictice/demo/Testing_ShipTo.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+@SpringBootTest
+public class Testing_ShipTo {
+    @Autowired
+    private WarehouseRepo WhR;
+    @Autowired
+    private ShipToRepo sTR;
+    @Test
+    public void addShipTo(){
+        ShipTo st=new ShipTo();
+        Warehouse wh = WhR.findById(Long.valueOf("3")).get();
```
Comment (metlaivan) [inline src/test/java/com/summer_prictice/demo/Testing_ShipTo.java:28]: Long.valueOf("3") -> 3L

File: src/test/java/com/summer_prictice/demo/Testing_ShipTo.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import java.time.LocalDate;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class Testing_ShipTo {
+    @Autowired
+    private WarehouseRepo WhR;
```
Comment (metlaivan) [inline src/test/java/com/summer_prictice/demo/Testing_ShipTo.java:22]: Please use the full name field.

File: src/test/java/com/summer_prictice/demo/Testing_ShipTo.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        LocalDate date_up =  LocalDate.of(2012,9,8);
+        st.setCreatedat(date_cr);
+        st.setCreatedby("Dalian");
+        st.setUpdatedat(date_up);
+        st.setUpdatedby("Less");
+        st.setName("Cars");
+        st.setAddress("Suite 12");
+        st.setWarehouseId(wh);
+        sTR.save(st);
+        assertEquals(sTR.findById(Long.valueOf("4")).get().getCreatedat(),date_cr);
```
Comment (metlaivan) [inline src/test/java/com/summer_prictice/demo/Testing_ShipTo.java:39]: Why 4.

File: src/test/java/com/summer_prictice/demo/Testing_ShipTo.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        assertTrue(sTR.findById(Long.valueOf("4")).isPresent());
```
Comment (metlaivan) [inline src/test/java/com/summer_prictice/demo/Testing_ShipTo.java:40]: never execute.

File: src/test/java/com/summer_prictice/demo/Testing_Warehouse.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class Testing_Warehouse {
+    @Autowired
+    private DistributorRepository dsR;
+    @Autowired
+    private WarehouseRepo WhR;
+    @Test
+    public void addWarehouse(){
+        Warehouse WT=new Warehouse();
```
Comment (metlaivan) [inline src/test/java/com/summer_prictice/demo/Testing_Warehouse.java:27]: please use normal variable names.

File: src/test/java/com/summer_prictice/demo/Testing_distributor.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.time.LocalDate;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class Testing_distributor {
```
Comment (metlaivan) [inline src/test/java/com/summer_prictice/demo/Testing_distributor.java:18]: format the code

**Review by metlaivan — CHANGES_REQUESTED** [review]
Take into account.!!!!!!!!!!
https://docs.oracle.com/javase/tutorial/java/nutsandbolts/variables.html
https://google.github.io/styleguide/javaguide.html#s5-naming

**Comment by metlaivan** [comment]
Please check the description of the task. Do you use all technologies from the list?

File: pom.xml [hunk 13]
```diff
@@ -49,10 +45,31 @@
...
+            <groupId>com.github.database-rider</groupId>
+            <artifactId>rider-spring</artifactId>
+            <version>1.12.0</version>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>
         <plugins>
+            <plugin>
```
Comment (metlaivan) [inline pom.xml:68]: What do you do with that plugin?

**Review by metlaivan — APPROVED** [review]

**Comment by metlaivan** [comment]
You should put the test in the same package as tested class.

File: pom.xml (hunk 13 above)
Comment (Viacheslav-Pavliuk) [reply pom.xml:68]: This is an old edit, I forgot to delete this.

**Pull request merged**